<launch>
    <arg name="camera_topic"       default="/camera/color/image_raw"/>
    <arg name="camera_info_topic"  default="/camera/color/camera_info"/>
    <arg name="markerSize"         default="0.5"/>
    
    <arg name="markerId"           default="0"/> 
    <arg name="ref_frame"          default="base_link"/> 

    <node pkg="tf2_ros" type="static_transform_publisher" name="base_to_camera_tf" 
          args="0.0 0.1 -0.1 0 0 3.1415926 base_link camera_color_optical_frame" />

    <node pkg="aruco_ros" type="single" name="aruco_single" output="screen">
        <remap from="/camera_info" to="$(arg camera_info_topic)" />
        <remap from="/image"       to="$(arg camera_topic)" />
        
        <remap from="/aruco_single/pose" to="/aruco_single/pose" />
        <remap from="/aruco_single/result" to="/aruco_single/result" />

        <param name="image_is_rectified" value="True"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="marker_id"          value="$(arg markerId)"/>
        <param name="reference_frame"    value="$(arg ref_frame)"/>   
        <param name="camera_frame"       value="camera_color_optical_frame"/>
        <param name="marker_frame"       value="aruco_marker_frame" />
    </node>
</launch>
